controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1131,6 +1137,363 @@ bool master_thread::handle_cmdu_1905_channel_selection_response(Socket *sd,
+{
+    LOG(DEBUG) << "Print Ap Metrics data map";
+    for (auto const &pair_agent : ap_metric_data) {
+        LOG(DEBUG) << std::endl
+                   << "  Ap Metrics from agent with bssid= "
+                   << network_utils::mac_to_string(pair_agent.first) << std::endl
+                   << "  channel_utilization =" << int(pair_agent.second.channel_utilization)
+                   << std::endl
+                   << "  number_of_stas_currently_associated="
+                   << int(pair_agent.second.number_of_stas_currently_associated) << std::endl
+                   << "  estimated_service_info_field_ac_be = 0x" << std::hex
+                   << int(pair_agent.second.estimated_service_info_field_ac_be[0]) << std::hex
+                   << int(pair_agent.second.estimated_service_info_field_ac_be[1]) << std::hex
+                   << int(pair_agent.second.estimated_service_info_field_ac_be[2]) << std::endl
+                   << "  estimated_service_info_field_ac_bk = 0x" << std::hex
+                   << int(pair_agent.second.estimated_service_info_field_ac_bk[0]) << std::hex
+                   << int(pair_agent.second.estimated_service_info_field_ac_bk[1]) << std::hex
+                   << int(pair_agent.second.estimated_service_info_field_ac_bk[2]) << std::endl
+                   << "  estimated_service_info_field_ac_vo = 0x" << std::hex
+                   << int(pair_agent.second.estimated_service_info_field_ac_vo[0]) << std::hex
+                   << int(pair_agent.second.estimated_service_info_field_ac_vo[1]) << std::hex
+                   << int(pair_agent.second.estimated_service_info_field_ac_vo[2]) << std::endl
+                   << "  estimated_service_info_field_ac_vi = 0x" << std::hex
+                   << int(pair_agent.second.estimated_service_info_field_ac_vi[0]) << std::hex
+                   << int(pair_agent.second.estimated_service_info_field_ac_vi[1]) << std::hex
+                   << int(pair_agent.second.estimated_service_info_field_ac_vi[2]);
+    }
+}
+
+bool master_thread::handle_cmdu_1905_link_metric_response(Socket *sd,
+                                                          ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+
+    auto mid = cmdu_rx.getMessageId();
+    LOG(INFO) << "Received LINK_METRIC_RESPONSE_MESSAGE, mid=" << std::dec << int(mid);
+
+    //getting reference for link metric data storage from db
+    auto &link_metric_data = database.get_link_metric_data_map();
+
+    //will hold new link metric data from Reporting Agent
+    son::node::link_metrics_data new_link_metric_data;
+
+    bool old_link_metrics_removed = false;
+
+    int tlvType;
+    // holds mac address of the reporting agent, used as link_metric_data key
+    sMacAddr reporting_agent_al_mac, neighbor_al_mac;
+
+    // parse all tlvs in cmdu and save ap metric data to database
+    while ((tlvType = cmdu_rx.getNextTlvType()) != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
```

Comment:
Now that we have `cmdu_rx.parse()`, we could use that instead. That also removes the need for old_link_metrics_removed.

It's a big change though, so nitpick.

Reply:
If you mean the auto parse - then it's not ready yet..